modern/1.21.5/src/main/kotlin/io/github/notenoughupdates/moulconfig/platform/ModernFontRenderer.kt

```diff
@@ -19,21 +22,54 @@ class ModernFontRenderer(val textRenderer: TextRenderer) :
     }
 
     override fun splitText(text: String, width: Int): List<String> {
-        val lines = ChatMessages.breakRenderedChatMessageLines(Text.literal(text), width, textRenderer)
+        val lines = ChatMessages.breakRenderedChatMessageLines(Text.literal(text), width, MinecraftClient.getInstance().textRenderer)
         val strings: MutableList<String> = ArrayList(lines.size)
-        for (iChatComponent in lines) {
-            var formattedText = ""
-            iChatComponent.accept { i, style, j ->
-                formattedText += j.toChar()
+
+        for (line in lines) {
+            var newLine = ""
+            var lastColor: TextColor? = null
+            var lastFormatting = ""
+            line.accept { index, style, codePoint ->
+                val color = style.color
+                if (color != lastColor) {
+                    lastColor = color
+                    lastFormatting = ""
+                    if (color != null) {
+                        newLine += color.toChatFormatting()
+                    }
+                }
+                val newFormatting = when {
+                    style.isBold -> "§l"
+                    style.isItalic -> "§o"
+                    style.isUnderlined -> "§n"
+                    style.isStrikethrough -> "§m"
+                    style.isObfuscated -> "§k"
+                    else -> ""
+                }
```

Comment:
This is incorrect, afaik. when selects the first branch instead of all branches that are true.